### PR TITLE
Media: Make filename checks less strict in 'wp_image_src_get_dimensions'

### PR DESCRIPTION
**Pull request by joemcgill**
This modifies the check for full size files so that only the basename is compared with the image `src` to avoid misses whenever the `src` path has been modified.

Fixes: https://core.trac.wordpress.org/ticket/52417
Props ianmjones

### CODE REVIEW

**Comment by mukeshpanchal27** [comment]
Hello,

As per the PHP Coding Standards document - https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#yoda-conditions can we use **Yoda Conditions** here? 

`if ( strpos( $image_src, wp_basename( $image_meta['file'] ) ) !== false ) {`

Replace to

`if ( false !=== strpos( $image_src, wp_basename( $image_meta['file'] ) ) ) {`

**Comment by hellofromtonya** [comment]
>As per the PHP Coding Standards document - https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#yoda-conditions can we use Yoda Conditions here?

Yoda Conditions doesn't apply in this case. Why? There's no variable on either side of the conditional expression. Instead, the expression compares a boolean literal `false` to a final result of the functions. 

IMO the code is well-formed and understandable.

File: src/wp-includes/media.php [hunk 1]
```diff
@@ -1604,7 +1604,7 @@ function wp_image_src_get_dimensions( $image_src, $image_meta, $attachment_id =
 	// Is it a full size image?
 	if (
 		isset( $image_meta['file'] ) &&
-		strpos( $image_src, $image_meta['file'] ) !== false
+		strpos( $image_src, wp_basename( $image_meta['file'] ) ) !== false
```
Comment (hellofromtonya) [inline src/wp-includes/media.php:1607]: Looks good @joemcgill 👍 👏

**Comment by joemcgill** [comment]
Fixed in https://core.trac.wordpress.org/changeset/50144

**Pull request closed without merging**